agent/06_agent_create_cluster.sh

```diff
@@ -339,6 +348,57 @@ function agent_pxe_boot() {
       done
 }
 
+# Configure the instances for booting off an iSCSI disk
+function agent_setup_iscsi_boot() {
+    set_file_acl
+
+    # The boot server is started since iSCSI uses a similar mechanism to
+    # retrieve the file for iSCSI boot
+    mkdir -p ${BOOT_SERVER_DIR}
+    setup_boot_server
+
+    # Start server iscsid
+    sudo systemctl enable --now iscsid
+
+    tmpiscsi_network=$(mktemp --tmpdir "iscsi-network--XXXXXXXXXX")
+    _tmpfiles="$_tmpfiles $tmpiscsi_network"
+
+    # Create the separate network used for iSCSI booting
+    agent_create_iscsi_network > ${tmpiscsi_network}
+    sudo virsh net-create ${tmpiscsi_network} 
+
+    # Setup firewall to allow access to the iscsi target
+    sudo firewall-cmd --zone libvirt --add-port=3260/tcp
+}
+
+# Create the iscsi targets
+function agent_iscsi_targets() {
+    agent_iso=$(get_agent_iso)
+
+    for (( n=0; n<${2}; n++ ))
+      do
+          # Note that name use for target must not have an underscore
+          local name=${1}-${n}
+          iscsi_disk=${SCRIPTDIR}/"iscsi-${name}"
+          agent_create_iscsi_target ${name} ${agent_iso} ${iscsi_disk}
+          agent_create_iscsi_pxe_file ${name} ${BOOT_SERVER_DIR}
```

Comment:
the file `agent.x86_64-iscsi.ipxe` created by `agent_create_iscsi_pxe_file` will contain only the target for the last node, no?

Reply:
Our QE uses the the hostname to map the target to the nodes /cc @bkopilov

Reply:
Yes, good point. This has so far been focused on getting SNO to work but I will update it to handle multi-node per your suggestion and the work that Benny has done.

Reply:
IMO for multi node ,  if you want to handle a single iPXE server for all nodes try to use pxe script that use the hostname as a key for the iqn.
When u  create the target use the same hostname for the IQN.
....
sanboot --keep iscsi:192.168.145.1::::iqn.2024-11.com.example:${hostname}